Build issue breaking motoman/fanuc/internal drivers

Hello, I'm running Ubuntu 18.04 with ROS Melodic, I just starting seeing an issue where the simple message/industrial core is causing issues in all of my motoman, fanuc and internal arm(ros industrial driver based) drivers. 


Here is my output:

```
In file included from /opt/ros/melodic/include/simple_message/socket/tcp_socket.h:36:0,
                 from /opt/ros/melodic/include/simple_message/socket/tcp_client.h:36,
                 from /opt/ros/melodic/include/industrial_robot_client/joint_trajectory_interface.h:44,
                 from /opt/ros/melodic/include/industrial_robot_client/joint_trajectory_streamer.h:36,
                 from /home/akash/test_ws/src/fanuc/fanuc_driver/src/fanuc_joint_streamer_node.cpp:41:
/opt/ros/melodic/include/simple_message/socket/simple_socket.h:196:3: error: ‘sockaddr_in’ does not name a type; did you mean ‘caddr_t’?
   sockaddr_in sockaddr_;
   ^~~~~~~~~~~
   caddr_t
/opt/ros/melodic/include/simple_message/socket/simple_socket.h: In constructor ‘industrial::simple_socket::SimpleSocket::SimpleSocket()’:
/opt/ros/melodic/include/simple_message/socket/simple_socket.h:150:19: error: ‘class industrial::simple_socket::SimpleSocket’ has no member named ‘sockaddr_’; did you mean ‘sock_handle_’?
     memset(&this->sockaddr_, 0, sizeof(this->sockaddr_));
                   ^~~~~~~~~
                   sock_handle_
/opt/ros/melodic/include/simple_message/socket/simple_socket.h:150:46: error: ‘class industrial::simple_socket::SimpleSocket’ has no member named ‘sockaddr_’; did you mean ‘sock_handle_’?
     memset(&this->sockaddr_, 0, sizeof(this->sockaddr_));
                                              ^~~~~~~~~
                                              sock_handle_
make[2]: *** [CMakeFiles/fanuc_driver_motion_streaming_interface_bswap.dir/src/fanuc_joint_streamer_node.cpp.o] Error 1
make[1]: *** [CMakeFiles/fanuc_driver_motion_streaming_interface_bswap.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
In file included from /opt/ros/melodic/include/simple_message/socket/tcp_socket.h:36:0,
                 from /opt/ros/melodic/include/simple_message/socket/tcp_client.h:36,
                 from /opt/ros/melodic/include/industrial_robot_client/robot_state_interface.h:41,
                 from /home/akash/test_ws/src/fanuc/fanuc_driver/src/fanuc_robot_state_node.cpp:41:
/opt/ros/melodic/include/simple_message/socket/simple_socket.h:196:3: error: ‘sockaddr_in’ does not name a type; did you mean ‘caddr_t’?
   sockaddr_in sockaddr_;
   ^~~~~~~~~~~
   caddr_t
/opt/ros/melodic/include/simple_message/socket/simple_socket.h: In constructor ‘industrial::simple_socket::SimpleSocket::SimpleSocket()’:
/opt/ros/melodic/include/simple_message/socket/simple_socket.h:150:19: error: ‘class industrial::simple_socket::SimpleSocket’ has no member named ‘sockaddr_’; did you mean ‘sock_handle_’?
     memset(&this->sockaddr_, 0, sizeof(this->sockaddr_));
                   ^~~~~~~~~
                   sock_handle_
/opt/ros/melodic/include/simple_message/socket/simple_socket.h:150:46: error: ‘class industrial::simple_socket::SimpleSocket’ has no member named ‘sockaddr_’; did you mean ‘sock_handle_’?
     memset(&this->sockaddr_, 0, sizeof(this->sockaddr_));
                                              ^~~~~~~~~
                                              sock_handle_
make[2]: *** [CMakeFiles/fanuc_driver_robot_state.dir/src/fanuc_robot_state_node.cpp.o] Error 1
make[1]: *** [CMakeFiles/fanuc_driver_robot_state.dir/all] Error 2
In file included from /opt/ros/melodic/include/simple_message/socket/tcp_socket.h:36:0,
                 from /opt/ros/melodic/include/simple_message/socket/tcp_client.h:36,
                 from /opt/ros/melodic/include/industrial_robot_client/robot_state_interface.h:41,
                 from /home/akash/test_ws/src/fanuc/fanuc_driver/src/fanuc_robot_state_node.cpp:41:
/opt/ros/melodic/include/simple_message/socket/simple_socket.h:196:3: error: ‘sockaddr_in’ does not name a type; did you mean ‘caddr_t’?
   sockaddr_in sockaddr_;
   ^~~~~~~~~~~
   caddr_t
/opt/ros/melodic/include/simple_message/socket/simple_socket.h: In constructor ‘industrial::simple_socket::SimpleSocket::SimpleSocket()’:
/opt/ros/melodic/include/simple_message/socket/simple_socket.h:150:19: error: ‘class industrial::simple_socket::SimpleSocket’ has no member named ‘sockaddr_’; did you mean ‘sock_handle_’?
     memset(&this->sockaddr_, 0, sizeof(this->sockaddr_));
                   ^~~~~~~~~
                   sock_handle_
/opt/ros/melodic/include/simple_message/socket/simple_socket.h:150:46: error: ‘class industrial::simple_socket::SimpleSocket’ has no member named ‘sockaddr_’; did you mean ‘sock_handle_’?
     memset(&this->sockaddr_, 0, sizeof(this->sockaddr_));
                                              ^~~~~~~~~
                                              sock_handle_
make[2]: *** [CMakeFiles/fanuc_driver_robot_state_bswap.dir/src/fanuc_robot_state_node.cpp.o] Error 1
make[1]: *** [CMakeFiles/fanuc_driver_robot_state_bswap.dir/all] Error 2
In file included from /opt/ros/melodic/include/simple_message/socket/tcp_socket.h:36:0,
                 from /opt/ros/melodic/include/simple_message/socket/tcp_client.h:36,
                 from /opt/ros/melodic/include/industrial_robot_client/joint_trajectory_interface.h:44,
                 from /opt/ros/melodic/include/industrial_robot_client/joint_trajectory_streamer.h:36,
                 from /home/akash/test_ws/src/fanuc/fanuc_driver/src/fanuc_joint_streamer_node.cpp:41:
/opt/ros/melodic/include/simple_message/socket/simple_socket.h:196:3: error: ‘sockaddr_in’ does not name a type; did you mean ‘caddr_t’?
   sockaddr_in sockaddr_;
   ^~~~~~~~~~~
   caddr_t
/opt/ros/melodic/include/simple_message/socket/simple_socket.h: In constructor ‘industrial::simple_socket::SimpleSocket::SimpleSocket()’:
/opt/ros/melodic/include/simple_message/socket/simple_socket.h:150:19: error: ‘class industrial::simple_socket::SimpleSocket’ has no member named ‘sockaddr_’; did you mean ‘sock_handle_’?
     memset(&this->sockaddr_, 0, sizeof(this->sockaddr_));
                   ^~~~~~~~~
                   sock_handle_
/opt/ros/melodic/include/simple_message/socket/simple_socket.h:150:46: error: ‘class industrial::simple_socket::SimpleSocket’ has no member named ‘sockaddr_’; did you mean ‘sock_handle_’?
     memset(&this->sockaddr_, 0, sizeof(this->sockaddr_));
                                              ^~~~~~~~~
                                              sock_handle_
make[2]: *** [CMakeFiles/fanuc_driver_motion_streaming_interface.dir/src/fanuc_joint_streamer_node.cpp.o] Error 1
make[1]: *** [CMakeFiles/fanuc_driver_motion_streaming_interface.dir/all] Error 2
```

Sorry for the big file dump, I just am not sure what I could do to fix this. I tried getting ros industrial from source on melodic devel and it still fails.

--- a/simple_message/include/simple_message/log_wrapper.h
+++ b/simple_message/include/simple_message/log_wrapper.h
@@ -32,7 +32,8 @@
 #ifndef LOG_WRAPPER_H_
 #define LOG_WRAPPER_H_
 
-#ifdef SIMPLE_MESSAGE_USE_ROS
+// remove ROS after Melodic (bw compat for #262)
+#if defined(SIMPLE_MESSAGE_USE_ROS) || defined(ROS)
 #include "ros/ros.h"
 #endif
 
@@ -55,7 +56,8 @@ namespace log_wrapper
     
 
 // Define ROS if this library will execute under ROS
-#ifdef SIMPLE_MESSAGE_USE_ROS
+// remove ROS after Melodic (bw compat for #262)
+#if defined(SIMPLE_MESSAGE_USE_ROS) || defined(ROS)
 
 // The LOG_COMM redirects to debug in ROS because ROS has
 // debug filtering tools that allow the communications messages

--- a/simple_message/include/simple_message/robot_status.h
+++ b/simple_message/include/simple_message/robot_status.h
@@ -60,7 +60,8 @@ enum RobotMode
   MANUAL = 1, AUTO = 2,
 };
 
-#ifdef SIMPLE_MESSAGE_USE_ROS
+// remove ROS after Melodic (bw compat for #262)
+#if defined(SIMPLE_MESSAGE_USE_ROS) || defined(ROS)
 int toROSMsgEnum(RobotModes::RobotMode mode);
 #endif
 
@@ -85,7 +86,8 @@ enum TriState
   TS_FALSE = 0,   TS_OFF = 0,  TS_DISABLED = 0,  TS_LOW = 0
 };
 
-#ifdef SIMPLE_MESSAGE_USE_ROS
+// remove ROS after Melodic (bw compat for #262)
+#if defined(SIMPLE_MESSAGE_USE_ROS) || defined(ROS)
 int toROSMsgEnum(TriStates::TriState state);
 #endif
 

--- a/simple_message/include/simple_message/socket/simple_socket.h
+++ b/simple_message/include/simple_message/socket/simple_socket.h
@@ -42,7 +42,8 @@
 #include "smpl_msg_connection.h"
 #endif
 
-#ifdef SIMPLE_MESSAGE_LINUX
+// remove LINUXSOCKETS after Melodic (bw compat for #262)
+#if defined(SIMPLE_MESSAGE_LINUX) || defined(LINUXSOCKETS)
 
 #ifndef _WIN32
 #include "sys/socket.h"

--- a/simple_message/include/simple_message/socket/tcp_socket.h
+++ b/simple_message/include/simple_message/socket/tcp_socket.h
@@ -40,7 +40,8 @@
 #include "shared_types.h"
 #endif
 
-#ifdef SIMPLE_MESSAGE_LINUX
+// remove LINUXSOCKETS after Melodic (bw compat for #262)
+#if defined(SIMPLE_MESSAGE_LINUX) || defined(LINUXSOCKETS)
 #ifndef WIN32
 #include "sys/socket.h"
 #include "netdb.h"

--- a/simple_message/include/simple_message/socket/udp_socket.h
+++ b/simple_message/include/simple_message/socket/udp_socket.h
@@ -42,7 +42,8 @@
 #include "smpl_msg_connection.h"
 #endif
 
-#ifdef SIMPLE_MESSAGE_LINUX
+// remove LINUXSOCKETS after Melodic (bw compat for #262)
+#if defined(SIMPLE_MESSAGE_LINUX) || defined(LINUXSOCKETS)
 #ifndef WIN32
 #include "sys/socket.h"
 #include "netdb.h"

--- a/simple_message/src/message_manager.cpp
+++ b/simple_message/src/message_manager.cpp
@@ -38,7 +38,8 @@
 #include "simple_message.h"
 #endif
 
-#ifdef SIMPLE_MESSAGE_USE_ROS
+// remove ROS after Melodic (bw compat for #262)
+#if defined(SIMPLE_MESSAGE_USE_ROS) || defined(ROS)
 #include "ros/ros.h"
 #else
 #include "unistd.h"
@@ -187,7 +188,8 @@ void mySleep(int sec)
 void MessageManager::spin()
 {
   LOG_INFO("Entering message manager spin loop");
-#ifdef SIMPLE_MESSAGE_USE_ROS
+// remove ROS after Melodic (bw compat for #262)
+#if defined(SIMPLE_MESSAGE_USE_ROS) || defined(ROS)
   while (ros::ok())
 #else
   while (true)

--- a/simple_message/src/robot_status.cpp
+++ b/simple_message/src/robot_status.cpp
@@ -38,7 +38,8 @@
 #include "log_wrapper.h"
 #endif
 
-#ifdef SIMPLE_MESSAGE_USE_ROS
+// remove ROS after Melodic (bw compat for #262)
+#if defined(SIMPLE_MESSAGE_USE_ROS) || defined(ROS)
 // Files below used to translate between ROS messages enums and
 // enums defined in this file
 #include "industrial_msgs/RobotMode.h"
@@ -55,7 +56,8 @@ namespace robot_status
 namespace RobotModes
 {
 
-#ifdef SIMPLE_MESSAGE_USE_ROS
+// remove ROS after Melodic (bw compat for #262)
+#if defined(SIMPLE_MESSAGE_USE_ROS) || defined(ROS)
 
 int toROSMsgEnum(RobotModes::RobotMode mode)
 {
@@ -83,7 +85,8 @@ int toROSMsgEnum(RobotModes::RobotMode mode)
 namespace TriStates
 {
 
-#ifdef SIMPLE_MESSAGE_USE_ROS
+// remove ROS after Melodic (bw compat for #262)
+#if defined(SIMPLE_MESSAGE_USE_ROS) || defined(ROS)
 
 int toROSMsgEnum(TriStates::TriState state)
 {

